main.py

```diff
@@ -93,24 +180,31 @@ def should_nag(message: discord.Message) -> bool:
     3. Embed is safe and there is a video
     """
     if not re.search(r"(//|mobile\.)(twitter|x)\.com", message.content):
-        return False
+        return None
     if not message.embeds:
-        return True
+        return NagType.FULL
     # sensitive check _must_ be done before other types
     # to prevent mixing NSFW and SFW links from embedding
     if any(_is_sensitive_tweet_embed(em) for em in message.embeds):
         channel = message.channel
         if channel.type in (discord.ChannelType.private, discord.ChannelType.group):
             # always allow NSFW in (group) DMs
-            return True
-        if hasattr(channel, "nsfw"):
+            return NagType.FULL
+        if hasattr(channel, "nsfw") and getattr(channel, "nsfw") is True:
```

Comment:
`is True` seems redundant here?

Reply:
Changed logic to type-narrow using `isinstance`